torch/nn/parallel/distributed.py

```diff
@@ -661,6 +684,20 @@ def __init__(
             os.environ.get("PYTORCH_DDP_USE_SIDE_STREAM", "1") == "1"
         )
 
+        # Initialize gradient buffers and register all reduce hook
+        self._grad_buffer = None
+        self._grad_views = []
+        self._use_vanilla_module = False
+        if len(self._delay_all_reduce_params) != 0:
+            self._register_delay_all_reduce_hook(
+                bucket_cap_mb=bucket_cap_mb,
+                process_group=self.process_group,
+                param_to_hook_all_reduce=param_to_hook_all_reduce,
+                device_ids=device_ids,
+            )
+            if self._use_vanilla_module:
+                return
```

Comment:
nit: seems this is not needed?

Reply:
When we are registering delay all reduce hook, we will check whether all parameters are specified in delay_all_reduce_named_params. If yes, we don't need DDP to construct reducer or other facilities.